### PR TITLE
test: added tests for api

### PR DESCRIPTION
**Pull request by DmitriyLewen**

### CODE REVIEW

File: router/api_test.go [hunk 1]
```diff
@@ -67,11 +106,350 @@ func TestListOutput(t *testing.T) {
...
+   package postee
+   result:=input
+
+outputs:
+- name: my-slack
+  type: slack
+  enable: true
+  url: https://hooks.slack.com/services/ABCDF/1234/TTT`
+
+func runFunc(funcName, cfgPath, dbPath, tenantName, psqlUrl string) error {
```
Comment (AndreyLevchenko) [inline router/api_test.go:371]: no need to swich here. you can assign function to variables and then add function references to test data

File: router/router.go [hunk 2]
```diff
@@ -156,11 +156,11 @@ func (ctx *Router) Send(data []byte) {
 	ctx.queue <- data
 }
 func (ctx *Router) deleteTemplate(name string, removeFromRoutes bool) error {
-	_, ok := ctx.outputs[name]
```
Comment (AndreyLevchenko) [inline router/router.go:159]: nice catch :)

File: router/api.go [hunk 3]
```diff
@@ -88,7 +88,10 @@ func AddOutput(output *data.OutputSettings) error {
 	return Instance().addOutput(output)
 }
 func UpdateOutput(output *data.OutputSettings) error {
-	Instance().deleteOutput(output.Name, false)
+	err := Instance().deleteOutput(output.Name, false)
```
Comment (AndreyLevchenko) [inline router/api.go:91]: pull latest changes from `postee-as-api-2` make sure `golangci-lint run` returns no errors.
see: https://github.com/golangci/golangci-lint

**Review by AndreyLevchenko — CHANGES_REQUESTED** [review]

File: router/api.go (hunk 3 above)
Comment (DmitriyLewen) [reply router/api.go:93]: Done

File: router/api_test.go (hunk 1 above)
Comment (DmitriyLewen) [reply router/api_test.go]: Done

File: router/api_test.go [hunk 4]
```diff
@@ -67,11 +106,350 @@ func TestListOutput(t *testing.T) {
...
+	assert.Equal(t, "*regoservice.regoEvaluator", fmt.Sprintf("%T", Instance().templates["legacy"]), "ScanRenderer is updated")
+
+	err = UpdateTemplate(&data.Template{Name: "badName"})
+	if err != nil && errors.Is(err, expectedError) {
+		t.Errorf("unexpected error, expected: %v, got: %v", expectedError, err)
+	}
+}
+
+func TestListTemplate(t *testing.T) {
+	Instance().cleanInstance()
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply router/api_test.go]: Done

File: router/api_test.go [hunk 5]
```diff
@@ -67,11 +106,350 @@ func TestListOutput(t *testing.T) {
...
+	defer Instance().cleanInstance()
+
+	AddRoute(inputRoute)
+	assert.Equal(t, 1, len(Instance().inputRoutes), "one route expected")
+	assert.Contains(t, Instance().inputRoutes, "my-route")
+	assert.Equal(t, "my-route", Instance().inputRoutes["my-route"].Name, "check name failed")
+	assert.Equal(t, "*routes.InputRoute", fmt.Sprintf("%T", Instance().inputRoutes["my-route"]), "check name failed")
+}
+
+func TestDeleteRoute(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply router/api_test.go:173]: Done

**Review by AndreyLevchenko — APPROVED** [review]

**Pull request merged**